tests/generator/generator.cpp

```diff
@@ -0,0 +1,1230 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "test_utils.h"
+
+#include "gtest/gtest.h"
+
+#include <libcellml>
+
+static const std::string EMPTY_STRING;
+
+TEST(Generator, emptyModel)
+{
+    libcellml::ModelPtr model = std::make_shared<libcellml::Model>();
+    libcellml::Generator generator;
+
+    generator.processModel(model);
+
+    EXPECT_EQ(size_t(0), generator.errorCount());
+
+    EXPECT_EQ(libcellml::Generator::ModelType::UNKNOWN, generator.modelType());
+
+    EXPECT_EQ(size_t(0), generator.stateCount());
+    EXPECT_EQ(size_t(0), generator.variableCount());
+
+    EXPECT_EQ(nullptr, generator.voi());
+    EXPECT_EQ(nullptr, generator.state(0));
+    EXPECT_EQ(nullptr, generator.variable(0));
+
+    EXPECT_EQ(EMPTY_STRING, generator.interfaceCode());
+    EXPECT_EQ(EMPTY_STRING, generator.implementationCode());
+}
+
+TEST(Generator, initialized_variable_of_integration)
```

Comment:
Shouldn't have underscores in these test names.

Reply:
Good point (that's what happens when you copy/paste). Consider it fixed in a commit soon to be pushed...